### DISCUSSION TITLE
useFieldArray Update Only Selected Field

### DISCUSSION THREAD

**Discussion by hjaber** [react-hook-form/react-hook-form#7616] [Q&A] [answered]
When using the update method from useFieldArray, how you can you only update the specific node?

In the example below, my goal is to only update the fistName and leaving lastName as is. The below code does update firstName appropriately to "Mr. Bill" but changes the lastName to a blank field.

```
<input {...register(`test.${index}.firstName`)} />
<input {...register(`test.${index}.lastName`)} />
update(index, { firstName: "Mr. Bill" })}
```

Does updating a single field require copying it first?

```
const prevLastNameValue = getValue(`test.${index}.lastName`) 
update(index, { firstName: "Mr. Bill", lastName: prevLastNameValue })}
```

https://codesandbox.io/s/react-hook-form-usefieldarray-update-selected-field-only-j1uch?file=/src/index.js

Thank you for your time

**Comment by bluebill1049** [comment, marked as answer]
`setValue` ?

**Reply by hjaber** [reply]
Thank Bill! So fast. 

I read the [docs](https://react-hook-form.com/api/useform/setvalue) wrong, I interpreted that setValue was being depreciated with useFieldArray so I was avoiding setValue but it is a simple solution.

> **Important**: use replace from useFieldArray instead, update entire field array with setValue will be removed in the next major version.

> - When used with field array:
> - It's recommend to use field array's methods such as replace/update instead of setValue.

**Reply by bluebill1049** [reply]
The next major version will only affect update an entire field array with `setValue`, semantically `replace` is much better than making `setValue` do everything.

**Reply by bluebill1049** [reply]
thank you for sponsoring https://github.com/react-hook-form/react-hook-form/commit/2251b865a2970668d8001571b136bfbfd1e2a608
